src/client/pkg/tracing/tracing.go

```diff
@@ -0,0 +1,187 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	otgrpc "github.com/opentracing-contrib/go-grpc"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go"
+	jaegercfg "github.com/uber/jaeger-client-go/config"
+	"google.golang.org/grpc"
+)
+
+// JaegerServiceName is the service name used when the client reports traces
+// to Jaeger
+const JaegerServiceName = "pachd"
+
+// jaegerOnce is used to ensure that the Jaeger tracer is only initialized once
+var jaegerOnce sync.Once
+
+var enableTracing bool
+
+// EnableTracing causes our opentracing GRPC interceptor to attach traces to
+// outgoing RPCs (once an RPC has a trace attached to it, the trace will cause
+// downstream RPCs to be traced as well). This is called by pachctl (controlled
+// by a flag), but not by pachd, so that the only way to trace a Pachyderm RPC
+// is by setting the appropriate flag in pachctl
+func EnableTracing(enabled bool) {
+	enableTracing = enabled
+}
```

Comment:
Should this be set through an environment variable?

Reply:
Done—I went back and forth on this a bit, because I don't think users will want to trace every `pachctl` call they make over a period of time—specifically, you probably don't want to `export PACH_ENABLE_TRACING=true`, which was my rationale behind making it a flag. But, having thought about it, users who include the Pachyderm client library in their code (particularly anyone who uses the vault plugin) could take advantage of this tracing code, except that they might not be able to set a flag on the outer binary. So I changed this to use the environment variable `PACH_ENABLE_TRACING`